src/components/library-app-5/screens/book-listing.js

```diff
@@ -0,0 +1,78 @@
+import axios from 'axios';
+import React, {useEffect, useState} from 'react';
+import {
+  FlatList,
+  Text,
+  View,
+  StyleSheet,
+  TouchableOpacity,
+  Alert,
+} from 'react-native';
+import {useDispatch, useSelector} from 'react-redux';
+import {libraryApiEndpoint as url} from '../../../constants/library';
+import navigationStrings from '../../../constants/navigation-strings';
+import {setBooks} from '../../../store/slices/books';
+
+const BookListing = ({navigation}) => {
+  const books = useSelector(state => state.books);
```

Comment:
bad practice. Alwways create const functions and use tem wrapped around selectors. otherwise useMemo